### PR TITLE
Walking trails https fix

### PR DESCRIPTION
**Pull request by martypowell**
- Updated the service to use the new services
- Fixed the query order clause to return the records in the proper order

### CODE REVIEW

File: js/page-specific/trails.js [hunk 1]
```diff
@@ -1,907 +1,982 @@
...
+                "<a href='https://www.google.com/maps/?daddr=${Lat},${Long}'>" +
+                "Get Directions</a><a href='#parkid${ID}-${OBJECTID}' class='js-infoWindow-more-info pull-right' " +
+                "data-object-id='${OBJECTID}' title='View More Information on ${NAME}'>More Information</a>" +
+                "<div id='attach'></div>" +
+                "</div>";
+        };
 
-        //Listen for click event on point
-        parkLayer.on("click", DisplayParkInfoWindow);
+        trailsTemplate = "<ul class='trails'>{{#.}}{{#attributes}}{{#NAME}}";
```
Comment (mxsnyder) [inline js/page-specific/trails.js]: Would you consider creating this template in a HTML file, and then loading it into the DOM via ajax?

File: js/page-specific/trails.js [hunk 2]
```diff
@@ -1,907 +1,982 @@
...
+            return "{{#.}}<a href='{{url}}' id='attach-{{objectId}}' class='btn btn-default trail-map object-{{objectId}}' title='View the Park Map'>" + label + "</a>{{/.}}";
+        };
 
-        trailLayer = new FeatureLayer("//gis.baltimorecountymd.gov/arcgis/rest/services/Apps/WalkingTrailsExternalViewer/MapServer/1", {
-            name: "trails",
-            mode: esri.layers.FeatureLayer.MODE_SNAPSHOT,
-            outFields: ["*"], //Return all fields from the layer
-            orderByFields: "NAME ASC"
-        });
+        window.myData = appData;
```
Comment (mxsnyder) [inline js/page-specific/trails.js]: Is using a global object necessary here?

File: js/page-specific/trails.js [hunk 3]
```diff
@@ -1,907 +1,982 @@
...
-            parkObjectIds = getFeatureProperty(results.features, "OBJECTID");
+                //Loop through each trail so that we can determine where it should go
+                for (var i = 0; i < trails.length; i++) {
 
-        //Get Trails and Then Append them to the Parks
-        GetTrails(parkIds);
+                    //Create html based off of the trail template
+                    var content = Mustache.render(trailTemplate, trails[i]),
+                        //Get the selector of the list item
+                        selector = "[data-park-id='" + trails[i].attributes.PARK_ID + "']",
```
Comment (mxsnyder) [inline js/page-specific/trails.js]: Would you give "selector" a more meaningful name? Like parkListItemSelector?

File: js/page-specific/trails.js [hunk 4]
```diff
@@ -1,907 +1,982 @@
...
-    /*Events*/
-    $(document).on("click", '.js-prev-btn', function (e) {
-        e.preventDefault();
-        queryRecordsByPage(pageInfo.currentPage - 1);
+            window.location.hash = pageInfo.currentPage === 1 ? "page-1" : "page-" + (pageInfo.currentPage);
+        });
 
-        window.location.hash = pageInfo.currentPage === 1 ? "page-1" : "page-" + (pageInfo.currentPage);
-    });
+        $(document).on("click", '.js-next-btn', function (e
```
Comment (mxsnyder) [inline js/page-specific/trails.js]: I think $this is a little misleading. In these cases, I usually call them $eventTarget or something similar.

Comment (mxsnyder) [reply js/page-specific/trails.js]: Also, did you mean to use evt.target or evt.currentTarget? This may be fine as-is, but it's worth thinking about.

Comment (mxsnyder) [inline js/page-specific/trails.js]: Would you cache $('.js-show-records') in this method since you seek it a few times?

Comment (mxsnyder) [inline js/page-specific/trails.js]: Extract handler to function. Also, you should be consistent on how you're naming event variables. I was just looking at where it was names "evt". One way around it is to name them "mouseEvent" or "keyboardEvent" and then you're relating meaningful info as well.

Comment (mxsnyder) [inline js/page-specific/trails.js]: One more - does this need to be global? It very well might, but it's worth asking.

File: js/page-specific/trails.js [hunk 5]
```diff
@@ -1,907 +1,982 @@
...
-
-        function decodeHTMLEntities(str) {
-            if (str && typeof str === 'string') {
-                // strip script/html tags
-                str = str.replace(/<script[^>]*>([\S\s]*?)<\/script>/gmi, '');
-                str = str.replace(/<\/?\w(?:[^"'>]|"[^"]*"|'[^']*')*>/gmi, '');
-                element.innerHTML = str;
-                str = element.textContent;
-                element.textContent = '';
+require([
```
Comment (mxsnyder) [inline js/page-specific/trails.js]: General comment - did you eslint this?

**Comment by martypowell** [comment]
Some of the review can/should be addressed in the future, but not at the moment. @danfox01 I will get in touch.

**Pull request closed without merging**